tests/smoke.rs

```diff
@@ -172,6 +172,10 @@ struct Project {
 
 impl Project {
     fn new(name: &'static str) -> Result<Self> {
+        Self::new_in(std::env::temp_dir(), name)
+    }
+
+    fn new_in(dir: PathBuf, name: &'static str) -> Result<Self> {
```

Comment:
So this will create a subdir and put everything into there, right? It won't work directly in `dir`?
Please add a doc comment explaining that. This had me confused for a second here.

Reply:
Right, that is a bit unintuitive. I'll add a comment.